docs/knowledgebase/learn-substrate/off-chain-features.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Off-Chain Features
+---
+
+## Overview
+
+There is often a need to query and/or process off-chain data before it can be included in the
+on-chain state. The conventional way of doing this is through oracles. Oracles are external services
+that typically listen to blockchain events and trigger tasks accordingly. When these tasks complete,
+their results are submitted back to the blockchain using transactions. While this approach works, it
+still has several flaws with respect to security, scalability, and infrastructure efficiency.
+
+To make the off-chain data integration secure and more efficient, Substrate provides off-chain
+features:
+
+- **Off-Chain Worker (OCW)** subsystem allows execution of long-running and possibly non-
+deterministic tasks (e.g. web requests, encryption/decryption and signing of data, random number
+generation, CPU-intensive computations, enumeration/aggregation of on-chain data, etc.) that could
+otherwise require longer than the block execution time.
+
+- **Off-Chain Storage** offers storage that is local to a Substrate node that can be accessed both by 
```

Comment:
Off-Chain Storage can only be written to by the runtime (not read) via indexing API, while off-chain workers can do both. I think it would be good to clarify that here. (see suggestion)

Reply:
Thanks for review